azure-function-app/massdriver.yaml

```diff
@@ -0,0 +1,188 @@
+        minimum_worker_count:
+          title: Minimum instances
+          description: Minimum number of instances for autoscale. (Enabling zone balancing multiplies this value by 3)
+          type: integer
+          minimum: 1
+          maximum: 30
+        maximum_worker_count:
+          title: Maximum instances
+          description: Maximum number of instances for autoscale.(Enabling zone balancing multiplies this value by 3)
+          type: integer
+          minimum: 1
+          maximum: 30
+        zone_balancing:
+          title: Zone balancing
+          description: Enable zone balancing for the App Service Plan. Cannot be changed after deployment. Only supported in East US.
+          $md.immutable: true
+          type: boolean
+          default: false
+    docker:
+      title: Docker Configuration
+      type: object
+      required:
+      - registry
+      - image
+      - tag
+      properties:
+        registry:
+          $md.immutable: true
+          title: Docker registry
+          # Massdriver automatically configures your app service to have access to your cloud container repositories.
+          # Just set the full URI here or remove this field and hard code it in your values.yaml file.
+          description: Docker registry to use for this application.
+          type: string
+        image:
+          title: Docker image
+          description: Docker image to use for this application.
+          type: string
+        tag:
+          title: Image Tag
+          $ref: https://raw.githubusercontent.com/massdriver-cloud/artifact-definitions/main/definitions/types/docker-image-tag.json
+          default: latest
+    dns:
+      title: DNS
+      description: DNS configuration
+      type: object
+      required:
+        - enable_dns
+      properties:
+        enable_dns:
+          title: Enable DNS
+          description: Enable DNS configuration for the application.
+          type: boolean
+          default: false
+      dependencies:
+        enable_dns:
+          oneOf:
+            - properties:
+                enable_dns:
+                  const: true
+                zone_id:
+                  title: DNS Zone name
+                  description: Azure DNS zone name
+                  type: string
+                subdomain:
+                  title: Subdomain
+                  description: "Subdomain for DNS zone. Examples: 'www', 'api', 'app'"
+                  type: string
+                  pattern: ^[-a-zA-Z0-9*._-]{1,63}$
+                  message:
+                    pattern: Must be a valid subdomain. Must contain only letters, numbers, dashes, underscores, periods, and asterisks.
+              required:
+                - zone_id
+                - subdomain
+            - properties:
+                enable_dns:
+                  const: false
+
+connections:
+  required:
+    - azure_service_principal
+    - azure_virtual_network
+  # <md- range $key, $art:= .Connections md>
```

Comment:
Are we going to uncomment before publishing?

Reply:
Ah, yes, sorry. Commented out for local testing.